File: api/src/org/ufomatics/archipelago/model/MetaTypeImpl.java
package org.ufomatics.archipelago.model;

import org.ufomatics.archipelago.model.MetaField;
import org.ufomatics.archipelago.model.MetaType;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

/**
 * @author <a href="mailto:rr0@rr0.org">Jerome Beau</a>
 * @version $revision$
 */
public class MetaTypeImpl implements MetaType, Serializable  {
    private String name;
    private Set<MetaField> fields = new HashSet<MetaField>();

    public MetaTypeImpl() {
    }

    public MetaTypeImpl(String name) {
        setName(name);
    }

    public void setName(String name) {
        this.name = name;
    }

    public void addField(MetaField metaField) {
        fields.add (metaField);
    }

    public MetaField createField() {
        return new MetaFieldImpl();
    }

    public String getName() {
        return name;
    }

    public void add(MetaField field) {
        fields.add (field);
    }

    public void setFields(Set<MetaField> fields) {
        this.fields = fields;
    }

    public Set<MetaField> getFields() {
        return fields;
    }

    public String toString() {
        return getName();
    }
}

File: src/archipelago/api/model/MetaTypeImpl.ts
import {MetaType} from "archipelago/api/model/MetaType"
import {MetaField} from "archipelago/api/model/MetaField"
import {MetaFieldImpl} from "archipelago/api/model/MetaFieldImpl"
import {JSerializable} from "ts-jsdk"

/**
 *
 */
export class MetaTypeImpl implements MetaType, JSerializable {
  fields = new Set<MetaField>()

  constructor(public name: string) {
  }

  setName(name: string): void {
    this.name = name
  }

  addField(metaField: MetaField): void {
    this.fields.add(metaField)
  }

  createField(name: string, type: MetaType): MetaField {
    return new MetaFieldImpl(name, type)
  }

  getName(): string {
    return this.name
  }

  add(field: MetaField): void {
    this.fields.add(field)
  }

  setFields(fields: Set<MetaField>): void {
    this.fields = fields
  }

  getFields(): Set<MetaField> {
    return this.fields
  }

  toString(): string {
    return this.getName()
  }
}

export const VOID = new class extends MetaTypeImpl {
}("void")
